wpeverest/user-registration
# Enhance - Limit username length and choice field.

### All Submissions:






### Changes proposed in this Pull Request:
Firstly, there was no setting that allowed the user to set a length limit for the username field. This PR provides this setting in the Advance Setting of Field Option for the username field.
Secondly, there was no setting that allowed the user to set a limit up to which choice can be selected in multiple-choice fields like checkbox and multi-select2. This PR provides this setting in the Advance Setting of Field Options of individual checkbox and multi-select2 field.
Thirdly, The jquery validation was not working in the edit-profile section of my account page. This PR solves this issue.

### How to test the changes in this Pull Request:
1. Navigate to form builder and drag username, checkbox & multi-select2 field.
2. Under the Advance Setting of Field Options of individual field you will find an option labelled as Limit Username Length. Set the desired number to be the limit of length up to which username is valid as per your requirement.
2. Under the Advance Setting of Field Options of individual choice field you will find an option labelled as Choice Limit. Set the desired number to be the limit up to which the choices can be selected in the respective field.
3. Now go to the registration form and edit-profile in the frontend and check if you can enter a username of length greater than the limit set and can choose/select more choices than the choice limit set for respective fields. 
4. Also verify if all validations are properly working in the registration form as well as the edit-profile page.

### Types of changes:




### Other information:




### Changelog entry

> Enhance - Limit username length and choice field.
